src/ServiceManager.php

```diff
@@ -59,19 +61,12 @@
  * @psalm-import-type FactoriesConfigurationType from ConfigInterface
  * @psalm-import-type InitializersConfigurationType from ConfigInterface
  * @psalm-import-type LazyServicesConfigurationType from ConfigInterface
- * @psalm-type ServiceManagerConfiguration = array{
- *     abstract_factories?: AbstractFactoriesConfigurationType,
- *     aliases?: array<string,string>,
- *     delegators?: DelegatorsConfigurationType,
- *     factories?: FactoriesConfigurationType,
- *     initializers?: InitializersConfigurationType,
- *     invokables?: array<string,string>,
- *     lazy_services?: LazyServicesConfigurationType,
- *     services?: array<string,object|array>,
- *     shared?:array<string,bool>,
- *     shared_by_default?:bool,
- *     ...
- * }
+ * @psalm-import-type ServiceManagerConfigurationType from ConfigInterface
```

Comment:
Not sure if we should alias `ServiceManagerConfiguration` here, mostly to ease downstream usage?

Reply:
Thats the import, has to be modified in the `ConfigInterface` tho.

One of the problems I do have with not having the `*Type` suffix is, that it looks in the rest of the file as it would be an interface or a class. Having the `Type`-suffix notation (I sometimes also prefer `ArrayShape`) makes it clear, that this is some sort of custom type.

So mainly I choose this for readability. But if you say that it is okay to remove it, I dont have a too strong opinion on this and open for changes.

Reply:
Thought about it again, it is `ServiceManagerConfigurationType` as of now but since upstream projects still have to modify the usage from `ConfigInterface` to `ServiceManager`, it should be possible to rename this as well.

- [ ] gonna rename this